etl/src/energuide/embedded/basement.py

```diff
@@ -349,14 +370,25 @@ def from_data(cls, basement: element.Element) -> 'Basement':
         walls = wall_from_data(wall_nodes[0], floors[0].perimeter.metres) if wall_nodes else []
         header = header_from_data(header_nodes[0]) if header_nodes else None
 
-        return Basement(
-            foundation_type=foundation_type,
-            label=basement.get_text('Label'),
-            configuration_type=basement.xpath('Configuration/@type')[0],
-            walls=walls,
-            floors=floors,
-            header=header,
-        )
+        try:
+            configuration_type = basement.get('Configuration/@type', str)
+            assert len(configuration_type) > 1
+        except AssertionError as exc:
+            raise InvalidEmbeddedDataTypeError(Basement, 'Missing configuration type') from exc
+        except ElementGetValueError as exc:
+            raise InvalidEmbeddedDataTypeError(Basement, 'Invalid configuration type') from exc
+
+        try:
+            return Basement(
+                foundation_type=foundation_type,
+                label=basement.get_text('Label'),
+                configuration_type=configuration_type,
+                walls=walls,
+                floors=floors,
+                header=header,
+            )
+        except ElementGetValueError as exc:
```

Comment:
Would be better to combine all the possible exception-raising logic into a single try/except block

Reply:
It would look nicer, but would result in less granular error messages.

Reply:
My point is that it includes a lot of stuff that couldn't raise these errors, which is undesirable if possible to avoid.